### PR TITLE
fix(metadata-editor): dont show filter dropdown for one visible instance

### PR DESCRIPTION
**Pull request by wpiesiak**
- Fixing bug when FilterInstancesDropdown was shown for only one instance where other added instances were hidden 
- move filtering of hidden templates and template instances to useSidebarMetadataFetcher
- add hidden template instance to mockEmptyMetadataInstances for visual tests

### CODE REVIEW

**Review by kajarosz — APPROVED** [review]

File: src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx [hunk 1]
```diff
@@ -141,6 +162,21 @@ describe('useSidebarMetadataFetcher', () => {
         expect(result.current.errorMessage).toBeNull();
     });
 
+    test('should fetch metadata and return only visible templates and template instances', async () => {
+        const { result } = setupHook();
+
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({
+                templates: [...mockTemplates, mockHiddenTemplate],
```
Comment (JChan106) [inline src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx:170]: Can we also have tests for only hidden template instances?

File: src/elements/content-sidebar/stories/__mocks__/MetadataSidebarRedesignedMocks.ts [hunk 2]
```diff
@@ -68,7 +68,19 @@ export const mockGlobalMetadataTemplates = {
 export const mockEmptyMetadataInstances = {
     url: `${apiV2Path}/files/${fileIdWithoutMetadata}/metadata`,
     response: {
-        entries: [],
+        entries: [
```
Comment (JChan106) [inline src/elements/content-sidebar/stories/__mocks__/MetadataSidebarRedesignedMocks.ts:71]: Why is this change needed?

Comment (wpiesiak) [reply src/elements/content-sidebar/stories/__mocks__/MetadataSidebarRedesignedMocks.ts:71]: Hidden instances should be omitted in MetadataSidebarRedesign. I've added it for extra layer of testing to verify this logic.

File: src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx (hunk 1 above)
Comment (wpiesiak) [reply src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx]: Change the approach and move the logic to `MetadataSidebarRedesign` and added tests to corresponding file

**Review by JChan106 — APPROVED** [review]

**Review by kajarosz — DISMISSED** [review]

**Review by kajarosz — APPROVED** [review]

**Review by JChan106 — APPROVED** [review]
